doc/SONiC_local_users_password_reset_hld.md

```diff
@@ -0,0 +1,300 @@
+        # Wait if the Warm/Fast boot is in progress
+        if swsscommon.RestartWaiter.isAdvancedBootInProgress(self.state_db_conn):
+            swsscommon.RestartWaiter.waitAdvancedBootDone()
+
+        self.platform_local_users_password_reset = get_platform_local_users_passwords_reset()
+
+    def get_feature_state(self):
+        feature_state = 'disabled'
+        try:
+            db = ConfigDBConnector()
+            db.connect(wait_for_init=True, retry_on=True)
+            feature_info = db.get_table(swsscommon.CFG_LOCAL_USERS_PASSWORDS_RESET)
+            feature_state = feature_info.get('feature', {}).get('state')
+            return True if feature_state == 'enabled' else False
+        except Exception:
+            syslog.syslog(syslog.LOG_ERR, "LocalUsersPasswordsReset: Failed to connect to DB")
+            raise
+
+    def start(self):
+        feature_enabled = self.get_feature_state()
+        should_trigger = self.platform_local_users_password_reset.should_trigger()
+
+        if should_trigger and feature_enabled:
+            syslog.syslog(syslog.LOG_INFO, "LocalUsersPasswordsReset: Long reboot detected, the system is resetting passwords to default")
+            self.platform_long_reboot_press.start()
+        else:
+            syslog.syslog(syslog.LOG_INFO, "LocalUsersPasswordsReset: Skipping..")
+
+
+def main():
+    LocalUsersConfigurationReset().start()
+```
+
+The feature is added to the init flow as long as we set the new ```ENABLE_LOCAL_USERS_PASSWORDS_RESET``` global variable and the exsiting ```CHANGE_DEFAULT_PASSWORD``` global variable in ```rules/config``` to ```y```.
+
+```
+# ENABLE_LOCAL_USERS_PASSWORDS_RESET - enable local users' passwords reset during init on switch
+ENABLE_LOCAL_USERS_PASSWORDS_RESET ?= y
+```
+
+###
+
+# Performance
+
+The ```reset-local-users-passwords.service``` is required to run as fast as possible, and it is expected to finish running in milliseconds **(150-300 milliseconds)** on long reset detected, and not delay the init flow more than that.
+
+# Flows
+
+###### Local users' password reset during init flow
+![](./reset-local-users-passwords-init-flow.png  "Local users' password reset during init")
+
+The defualt behavior is on long reset button press, the system will delete non default users and restore original passwords for default users and expire them, all done using linux commands
+
+# Configuration and management
+
+## CLI
+
+<!-- omit in toc -->
+## Command structure
+
+**User interface**:
+```
+config
+\-- local-users-passwords-reset
+    |-- state <enabled|disabled>
+
+show
+\-- local-users-passwords-reset
+```
+
+### Config command group
+**The following command set local-users-passwords-reset  feature state:**
+```bash
+config local-users-passwords-reset state <enabled|disabled>
+```
+
+### Show command
+**The following command display long-reset-button-press  configuration:**
+```bash
+root@sonic:/home/admin# show local-users-passwords-reset
```

Comment:
I would prefer the show command to show the last reboot reset button state, rather than just showing the configuration. Can you please publish the last reboot reset button state to state_db and use it to display in show, so that user will be aware if a long press is detected after the current boot?

Reply:
If show reboot-cause can cover it, I am fine with it